controller/workitem.go

```diff
@@ -352,7 +352,7 @@ func ConvertJSONAPIToWorkItem(ctx context.Context, method string, appl applicati
 			}, "assigning the work item to the root area of the space.")
 			err := appl.Spaces().CheckExists(ctx, spaceID)
 			if err != nil {
-				return errors.NewBadParameterError("space", spaceID).Expected("valid space ID")
```

Comment:
I've changed this because the error isn't for missing space.

Reply:
That is true but do you really think it is an internal error?

Reply:
Yes. Because the user is trying to fetch a `workitem` and `workitem` depends on a `space`.  The 
**ConvertJSONAPIToWorkItem** (https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitem.go#L260) is called at only 3 places([1],[2],[3]). Before all the three function calls, we validate that the `space exists` and then convert the workitem. If the `spaceid` turns out to be invalid while converting the workitem, it has to be an `internal server error` because the `space` was found in earlier stages.

Also, based on this[4], the `spaceid` can never be an invalid one. The `integrity constraint` would break in that case. 

This has to be an **Internal Server Error**.

1. https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitem.go#L120 
2. https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitems.go#L121
3. https://github.com/fabric8-services/fabric8-wit/blob/master/controller/workitems.go#L295
4. https://github.com/fabric8-services/fabric8-wit/blob/master/migration/sql-files/040-add-space-id-wi-wit-tq.sql#L5